app/src/main/java/com/abominableshrine/taptounlock/UnlockService.java

```diff
@@ -0,0 +1,218 @@
+ */
+public class UnlockService extends Service {
+    /* *****************************
+     * SECTION 1: GLOBAL VARIABLES
+     * ***************************** */
+    private final static Boolean DEBUG = AppConstants.DEBUG;
+    private boolean mBound;
+    private static Boolean running; // Boolean class to enable check for null
+
+    protected static UnlockService instance;
+
+    // Variables used for the communication to the TapPatternDetectorService
+    private Intent tapPatternDetectorServiceIntent;
+    private IBinder mIBinder;
+    private Messenger mUnlockServiceMessenger = null;
+    private ServiceConnection mUnlockServiceConnection = new ServiceConnection() {
+        public void onServiceConnected(ComponentName className, IBinder service) {
+            mUnlockServiceMessenger = new Messenger(service);
+            mBound = true;
+            mIBinder = mUnlockServiceMessenger.getBinder();
+        }
+
+        public void onServiceDisconnected(ComponentName className) {
+            mUnlockServiceMessenger = null;
+            mBound = false;
+            mIBinder = null;
+        }
+    };
+    private TapPattern mTapPattern = new TapPattern();
+    // The client class is used to enable simple communication to the TapPatternDetectorService
+    private UnlockServiceTapPatternDetectorClient mUnlockServiceTapPatternDetectorClient;
+
+    // Variables used for the ScreenOffBroadcastReceiver
+    private static ScreenOffBroadcastReceiver mScreenOffBroadcastReceiver;
+    private IntentFilter screenOffFilter;
+
+    /* *****************************
+     * SECTION 2: METHODS
+     * ***************************** */
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        instance = this;
+        running = true;
+
+        // Initialize communication to the TapPatternDetectorService
+        tapPatternDetectorServiceIntent = new Intent(this, TapPatternDetectorService.class);
+        bindService(tapPatternDetectorServiceIntent, mUnlockServiceConnection, Context.BIND_AUTO_CREATE);
+        mUnlockServiceTapPatternDetectorClient = new UnlockServiceTapPatternDetectorClient(mIBinder);
+
+        // Initialization of the ScreenOffBroadcastReceiver
+        mScreenOffBroadcastReceiver = new ScreenOffBroadcastReceiver();
+        screenOffFilter = new IntentFilter();
+        screenOffFilter.addAction(Intent.ACTION_SCREEN_OFF);
+        registerReceiver(mScreenOffBroadcastReceiver, screenOffFilter);
+
+        // Some TapPattern
+        mTapPattern.appendTap(TapPattern.DeviceSide.FRONT, 1000);
+        mTapPattern.appendTap(TapPattern.DeviceSide.FRONT, 1000);
+        mTapPattern.appendTap(TapPattern.DeviceSide.FRONT, 1000);
+        mUnlockServiceTapPatternDetectorClient.subscribe(mTapPattern);
+
+        return START_STICKY;
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        throw new UnsupportedOperationException("UnlockService.onBind(Intent intent) not implemented!");
```

Comment:
You could just return `null`. This seems to be the preferred practice if a service does not support binding.